zend_abstract_interface/zai_sapi/zai_sapi.h

```diff
@@ -78,13 +78,49 @@ bool zai_sapi_append_system_ini_entry(const char *key, const char *value);
  */
 bool zai_sapi_execute_script(const char *file);
 
-/* Returns true if 'msg' exactly matches the last error message from PHP
- * globals.
+/* Inserts a fake frame scoped to the 'Zai\noop' internal function into the
+ * active PHP execution stack. If there is no active execution context, the
+ * fake frame will be the first frame in the stack. If there is an existing
+ * stack active in the executor globals, the fake frame will be added to the
+ * top of the stack.
+ *
+ * Motivation: sometimes it is necessary to emulate code execution during
+ * runtime (e.g. a custom opcode handler) or to create an execution context
+ * for certain functionality (e.g. throw an exception when there is no active
+ * execution context). A fake frame provides an execution context for these
+ * cases.
```

Comment:
Thanks for documenting this. I saw the `.c` file first and my first immediate thought was, "Why on earth are we pushing fake frames?" So I jumped to the header and there it is.